ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/OkHttpClientProvider.java

```diff
@@ -0,0 +1,117 @@
+package org.cru.godtools.articles.aem.service;
+
+import android.os.Build;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Collections;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+import okhttp3.ConnectionSpec;
+import okhttp3.OkHttpClient;
+import okhttp3.TlsVersion;
+import timber.log.Timber;
+
+public class OkHttpClientProvider {
+
+    public static OkHttpClient.Builder createBuilder () {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        return getEnabledTLSSupportedBuilder(builder);
+    }
+
+    private static OkHttpClient.Builder getEnabledTLSSupportedBuilder(OkHttpClient.Builder builder) {
+        if (Build.VERSION.SDK_INT >= 21) {
+            return builder;
+        }
+
+        try {
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
+                    TrustManagerFactory.getDefaultAlgorithm());
+            trustManagerFactory.init((KeyStore) null);
+            TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+            if (trustManagers.length != 1 || !(trustManagers[0]instanceof X509TrustManager)) {
+                throw new IllegalStateException("Unexpected default trust managers: " + Arrays.toString(trustManagers));
+            }
+            X509TrustManager trustManager = (X509TrustManager) trustManagers[0];
+
+            builder.sslSocketFactory(new Tls12SocketFactory(), trustManager);
+
+            ConnectionSpec cs = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS).tlsVersions(TlsVersion.TLS_1_2).build();
+
+            builder.connectionSpecs(Collections.singletonList(cs));
```

Comment:
this logic is written to only support TLS 1.2, and disable all other protocols. This is not what we want. We only want to add TLS 1.2 to our list of supported protocols if the current OS supports it.

Reply:
OK let me work on that.

Reply:
you don't need to modify the connection spec at all, the OkHttp default connectionspec is fine